connector/ldap/ldap.go

```diff
@@ -330,12 +334,17 @@ func (c *ldapConnector) do(_ context.Context, f func(c *ldap.Conn) error) error
 	}
 	defer conn.Close()
 
-	// If bindDN and bindPW are empty this will default to an anonymous bind.
-	if err := conn.Bind(c.BindDN, c.BindPW); err != nil {
-		if c.BindDN == "" && c.BindPW == "" {
+	if c.AnonymousBind {
```

Comment:
Do we need to add a new option here? I mean, according to the commentary 
> // If bindDN and bindPW are empty this will default to an anonymous bind.

Dex should use an unauthenticated bind if the `binddn` and `bindpw` options are missing.

Maybe it is good to leave things that way and only change a clause?
```go
if c.BindDN == "" && c.BindPW == "" {
  if err := conn.UnauthenticatedBind(""); err != nil {			
    return fmt.Errorf("ldap: initial anonymous bind failed: %v", err)		
  }
}
...
```

About naming, both `UnauthenticatedBind` and `AnonymousBind` look fine to me.

Reply:
Hm, makes sense to me.